gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
+      batch.submit();
+      assertEqChangesIgnoreStatus(created, result.get());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+      zone.applyChangeRequest(CHANGE_DELETE_ZONE1);
+      created = zone.applyChangeRequest(CHANGE_ADD_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.START_TIME));
+      batch = DNS.batch();
+      result = batch.getChangeRequest(zone.name(), created.generatedId(),
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.START_TIME));
+      batch.submit();
+      assertEqChangesIgnoreStatus(created, result.get());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+      zone.applyChangeRequest(CHANGE_DELETE_ZONE1);
+      created = zone.applyChangeRequest(CHANGE_ADD_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ADDITIONS));
+      batch = DNS.batch();
+      result = batch.getChangeRequest(zone.name(), created.generatedId(),
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ADDITIONS));
+      batch.submit();
+      assertEqChangesIgnoreStatus(created, result.get());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+      // finishes with delete otherwise we cannot delete the zone
+      created = zone.applyChangeRequest(CHANGE_DELETE_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.DELETIONS));
+      batch = DNS.batch();
+      result = batch.getChangeRequest(zone.name(), created.generatedId(),
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.DELETIONS));
+      batch.submit();
+      assertEqChangesIgnoreStatus(created, result.get());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+    } finally {
+      clear();
+    }
+  }
+
+  @Test
+  public void testListChangesBatch() {
+    try {
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Page<ChangeRequest>> result = batch.listChangeRequests(ZONE1.name());
+      batch.submit();
+      try {
+        result.get();
+        fail("Zone does not exist yet");
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(404, ex.code());
+        assertFalse(ex.retryable());
+      }
+      // zone exists but has no changes
+      DNS.create(ZONE1);
+      batch = DNS.batch();
+      result = batch.listChangeRequests(ZONE1.name());
+      batch.submit();
+      ImmutableList<ChangeRequest> changes = ImmutableList.copyOf(result.get().iterateAll());
+      assertEquals(1, changes.size()); // default change creating SOA and NS
+      // zone has changes
+      ChangeRequest change = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1);
+      waitForChangeToComplete(ZONE1.name(), change.generatedId());
+      change = DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+      waitForChangeToComplete(ZONE1.name(), change.generatedId());
+      change = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1);
```

Comment:
You could probably cut the last add and delete to make 3 total changes and still test everything you want to, right?